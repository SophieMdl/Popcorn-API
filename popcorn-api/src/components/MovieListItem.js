import React from 'react'
import { URL } from '../store'

const MovieListItem = ({movie, changeCurrentMovie}) => {
  console.log('movie', changeCurrentMovie);
  return (
    <li className="movie-item row" key={movie.id} onClick={() => changeCurrentMovie(movie)} >
      <img className="col s4" alt={movie.title} src={`${URL.IMAGE_BASE}${movie.poster_path}`} />
      <h6 className="col s8">{movie.title}</h6>
    </li>
  )
}

export default MovieListItem
