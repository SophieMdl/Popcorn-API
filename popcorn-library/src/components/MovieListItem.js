import React from 'react'
import { URL } from '../url'

const MovieListItem = ({movie, changeCurrentMovie}) => {
  return (
    <li className="movie-item row" onClick={() => changeCurrentMovie(movie)} >
      <img className="col s4" alt={movie.title} src={`${URL.IMAGE_BASE}${movie.poster_path}`} />
      <h6 className="col s8">{movie.title}</h6>
    </li>
  )
}

export default MovieListItem
