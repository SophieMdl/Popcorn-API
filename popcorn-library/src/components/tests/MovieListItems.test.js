import React from 'react'
import ReactDOM from 'react-dom'
import MovieListItem from '../MovieListItem'
import renderer from 'react-test-renderer'

const movie = {
  title: 'movie Title',
  poster_path: '/dACJtXkABwEBdX3FUrpHgJdlu9r.jpg'
}

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<MovieListItem movie = {movie} />, div)
  ReactDOM.unmountComponentAtNode(div)
})

it('renders MovieListItem html correctly', () => {
  const tree = renderer
    .create(<MovieListItem movie = {movie} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
