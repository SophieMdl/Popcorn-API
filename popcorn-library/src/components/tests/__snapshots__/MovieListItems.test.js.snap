// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieListItem html correctly 1`] = `
<div
  className="movie-item row"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex={0}
>
  <img
    alt="movie Title"
    className="col s4"
    src="http://image.tmdb.org/t/p/w500//dACJtXkABwEBdX3FUrpHgJdlu9r.jpg"
  />
  <h6
    className="col s8"
  >
    movie Title
  </h6>
</div>
`;
